src/rrd_cgi.c

```diff
@@ -560,13 +559,13 @@ static char *rrdsetenv(
     const char **args)
 {
     if (argc >= 2) {
-        const size_t len = strlen(args[0]) + strlen(args[1]) + 2;
-        char *xyz = (char *) malloc(len);
+        char *xyz = NULL;
 
+        asprintf(&xyz, "%s=%s", args[0], args[1]);
```

Comment:
we have been avoiding asprintf since it is a GNU extension ... 

These functions are GNU extensions, not in C or POSIX.  They are also available under *BSD.  The FreeBSD implementation  sets  strp  to  NULL  on error.

How would folks runing on solaris, aix, hpux handle thins?